core/src/main/java/aima/core/search/basic/support/StateActionTimeLine.java

```diff
@@ -0,0 +1,31 @@
+package aima.core.search.basic.support;
```

Comment:
This type of utility is likely best not added to the 'core' module. The 'extra' module is likely better but I would recommend removing this kind of thing from an initial submission as it bloats the functionality prematurely.